drivers: spi: spi_nrfx_spim: Add CPOL handling on SCK pin

awojasinski opened the pull request:
Pin state after SPIM deinitialization is based on pinctrl configuration. On the other hand, CPOL is set during runtime. With the introduction of the power-optimized SPIM driver, it disables the peripheral instance once the transfer is completed. As a result, the GPIO takes control over the SCK pin and drives it based on pinctrl configuration which causes an invalid SCK state when the transaction is configured with CPOL (Clock Polarity).

To address this issue, a patch was introduced to the SPIM driver. Now, when a SPIM instance is configured with CPOL, the driver is setting in the runtime the correct state of the SCK pin.

The PR addresses concerns in the comment https://github.com/zephyrproject-rtos/zephyr/pull/61280#issuecomment-1669621931 placed in PR with another solution to CPOL handling on SCK pin.

Commit:
drivers: spi: spi_nrfx_spim: Add CPOL handling on SCK pin

Pin state after SPIM deinitialization is based on pinctrl configuration.
On the other hand, CPOL is set during runtime. With the introduction
of the power-optimized SPIM driver, it disables the peripheral instance
once the transfer is completed.
As a result, the GPIO takes control over the SCK pin and drives it
based on pinctrl configuration which causes an invalid SCK state
when the transaction is configured with CPOL (Clock Polarity).

To address this issue, a patch was introduced to the SPIM driver.
Now, when a SPIM instance is configured with CPOL,
the driver is setting in the runtime the correct state of the SCK pin.

Signed-off-by: Adam Wojasinski <adam.wojasinski@nordicsemi.no>

anangl commented on drivers/spi/spi_nrfx_spim.c:
```diff
@@ -116,13 +117,32 @@ static inline nrf_spim_bit_order_t get_nrf_spim_bit_order(uint16_t operation)
 	const struct spi_nrfx_config *dev_config = dev->config;
 	struct spi_context *ctx = &dev_data->ctx;
 	uint32_t max_freq = dev_config->max_freq;
+	uint32_t sck_pin = get_nrf_spim_sck_pin(dev_config->pcfg);
```
```suggestion
	uint32_t sck_pin = nrf_spim_sck_pin_get(&dev_config->spim);
```

anangl reviewed (dismissed):
Could you also provide the same correction in `spi_nrfx_spi.c`?

Commit:
drivers: spi: spi_nrfx_spi: Add CPOL handling on SCK pin

Pin state after SPI deinitialization is based on pinctrl configuration.
On the other hand, CPOL is set during runtime. When the SPI instance
is disabled GPIO takes control over SCK and drives it to state set
by pinctrl driver. This might causes an invalid SCK state
when the transaction is configured with CPOL (Clock Polarity).

To address this issue, a patch was introduced to the SPI driver.
Now, when a SPI instance is configured with CPOL,
the driver is setting in the runtime the correct state of the SCK pin.

Signed-off-by: Adam Wojasinski <adam.wojasinski@nordicsemi.no>

anangl commented:
I guess #61280 can be closed.
